### PR TITLE
potential optimization - bail out of `executeSubSelectedArray` calls

### PR DESCRIPTION
**Pull request by phryneas**
This just came up in #11669 and I wanted to try it out.

One thing to consider strongly here is that this returns the original empty array, so I'll have to investigate if that could make it out of the cache where users could accidentally start manipulating it (and if we see this as a problem).

We could have a "weak empty array mapping cache" somewhere to prevent that, but at that point I'm not sure if this optimization does make any sense at all.

### CODE REVIEW

File: src/cache/inmemory/readFromStore.ts [hunk 1]
```diff
@@ -403,15 +403,18 @@ export class StoreReader {
...
-              array: fieldValue,
-              enclosingRef,
-              context,
-            }),
-            resultName
-          );
+          fieldValue =
+            fieldValue.length === 0 ?
+              fieldValue
+            : handleMissing(
```
Comment (benjamn) [inline src/cache/inmemory/readFromStore.ts:409]: You could probably rearrange this so it only calls `handleMissing` and updates `fieldValue` if `fieldValue.length > 0`, but I leave that 🏌️ up to you.

**Review by benjamn — APPROVED** [review]
This looks safe to me. The `StoreReader` often returns `fieldValue` to application code without making a defensive copy, but frozen (in development), so copying `fieldValue` in this case seems unusual/unnecessary.

**Review by jerelmiller — APPROVED** [review]
Nice! 🎉

**Pull request merged**